### PR TITLE
chore: KoboldAI/koboldcpp updates

### PR DESCRIPTION
**Pull request by g4rg**
* update KoboldAI lite embedded UI to v97
* update behaviour to reflect koboldcpp - best effort (#42)

### CODE REVIEW

**Comment by AlpinDale** [comment]
Do we need to wait for upstream to support genkeys before merging this?

**Comment by g4rg** [comment]
No but it won't be used before the impersonated version matches the one this is implemented upstream.
Which means adding a bunch of other new stuff from intermediate versions to validation.
And it should be tested.

I'll do that in a bit.

**Comment by AlpinDale** [comment]
Let me know when this is ready for merge or if it needs more testing.

**Comment by g4rg** [comment]
This is now reworked and basic-tested to reflect latest changes in koboldcpp and lite.
It impersonates the latest koboldcpp version that supports aborting individual streams.
Other new features that aren't implemented at the moment (mirostat, grammar) are introduced as stub, which goes against @LostRuins's recommendation of only impersonating newer versions whose capabilities are supported.
So it's either no support for `/abort`, or faking support for features that are not implemented. I don't know what's preferred here.

**Comment by LostRuins** [comment]
Yeah it's strongly not recommended to impersonate a version you can't perfectly feature match, as the client will assume certain capabilities are there, and break/misbehave if they don't work correctly. For example, if it thinks you can do SSE, it will attempt to do so and not use the normal generate endpoint if SSE is enabled.

Btw @g4rg did you grab the klite.embd from kcpp? If you grabbed from the lite repo, you'll need to change the localflag value to true.

**Comment by g4rg** [comment]
> Btw @g4rg did you grab the klite.embd from kcpp? If you grabbed from the lite repo, you'll need to change the localflag value to true.
I took it from the lite repo and changed the localflag. I also changed the default tokenstreammode to 2 (sse) to make it actually use it by default. Polling only gives empty intermediate results, it amounts to transmitting the full response through the original request when it is complete. I guess it's possible to add polling like in kcpp in multiuser mode.
> Also genkeys are only useful in multiuser mode. Normal abort is supposed to work fine without a key, when not running multiuser with queueing, which I assume aphrodite doesn't support.
Aphrodite always needs a request id. If no genkey is provided, it must be randomly generated. I could add caching of the last id to be used for abort when no genkey is provided, but this can break, aborting the wrong generation if used with multiple clients that aren't using genkeys (e.g. horde scribe + singleuser ui).

In summary, I don't see a way to support singleuser mode like in kcpp without workarounds that

**Comment by LostRuins** [comment]
Okay, I'll just let you know the SSE streaming is not considered fully reliable and I wouldn't leave it on by default. It also does require a much newer browser to actually work (TransformStreams, WritableStreams), though there is some graceful fallbacks for unsupported browsers. 

In general I would still recommend against just spoofing the latest client version, unless you are prepared to implement/troubleshoot all the required features & endpoints.

**Comment by g4rg** [comment]
> @g4rg the changes from #139 will introduce conflicts here so I'd like to merge this PR before that one. Is it ready for merging? I ran a few tests and it works fine.

Yes, I think it's fine.
